roles/mysql_hardening/tasks/mysql_secure_installation.yml

```diff
@@ -83,25 +82,11 @@
     (mysql_distribution == "mariadb" and mysql_version.version.full is
     version('10.4.0', '<'))
 
-- name: Create a fact for users without password or authentication_string
-  ansible.builtin.set_fact:
-    users_wo_auth: "{{ mysql_users_wo_passwords_or_auth_string.query_result.0.0 | community.general.json_query('users') }}"
-  when:
-    - mysql_users_wo_passwords_or_auth_string.query_result is defined
-    - mysql_users_wo_passwords_or_auth_string.query_result != "" # noqa empty-string-compare
-
-- name: Create a fact for users without password
-  ansible.builtin.set_fact:
-    users_wo_auth: "{{ mysql_users_wo_passwords.query_result.0.0 | community.general.json_query('users') }}"
-  when:
-    - mysql_users_wo_passwords.query_result is defined
-    - mysql_users_wo_passwords.query_result != "" # noqa empty-string-compare
-
 - name: Ensure that there are no users without password or authentication_string
   community.mysql.mysql_query:
     query:
-      - DROP USER {{ users_wo_auth }}
+      - DROP USER {{ item.users }}
     login_unix_socket: "{{ login_unix_socket | default(omit) }}"
-  when:
-    - users_wo_auth is defined
-    - users_wo_auth != "" # noqa empty-string-compare
+  with_community.general.flattened:
+    - "{{ mysql_users_wo_passwords.query_result.0 | default([]) }}"
```

Comment:
The `.0` seems to be not necessary with `with_flattened`?

test-play:

```
- hosts: localhost
  tasks:
  - name: Get all users that have no authentication_string on MySQL version >= 5.7.6 or Mariadb version >= 10.4.0
    community.mysql.mysql_query:
      query: >
        SELECT CONCAT(QUOTE(USER), '@', QUOTE(HOST)) AS users
        FROM mysql.user
        WHERE (length(authentication_string)=0
               OR authentication_string="")
          AND USER NOT IN ('mysql.sys',
                           'mysqlxsys',
                           'mariadb.sys');
      login_unix_socket: "{{ login_unix_socket | default(omit) }}"
    register: mysql_users_wo_passwords_or_auth_string

  - debug:
      msg: "{{ item.users }};"
    with_community.general.flattened:
      - "{{ mysql_users_wo_passwords_or_auth_string.query_result.0 | default([]) }}"

  - debug:
      msg: "{{ item.users }};"
    with_community.general.flattened:
      - "{{ mysql_users_wo_passwords_or_auth_string.query_result | default([]) }}"

```

result:

```
root@instance:/# ansible-playbook test.yml

PLAY [localhost] ****************************************************************************************************************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************************************************************************************************
ok: [localhost]

TASK [Get all users that have no authentication_string on MySQL version >= 5.7.6 or Mariadb version >= 10.4.0] ******************************************************************************************************
ok: [localhost]

TASK [debug] ********************************************************************************************************************************************************************************************************
ok: [localhost] => (item={'users': "'user'@'127.0.0.1'"}) => {
    "msg": "'user'@'127.0.0.1';"
}
ok: [localhost] => (item={'users': "'user'@'delete'"}) => {
    "msg": "'user'@'delete';"
}

TASK [debug] ********************************************************************************************************************************************************************************************************
ok: [localhost] => (item={'users': "'user'@'127.0.0.1'"}) => {
    "msg": "'user'@'127.0.0.1';"
}
ok: [localhost] => (item={'users': "'user'@'delete'"}) => {
    "msg": "'user'@'delete';"
}

PLAY RECAP **********************************************************************************************************************************************************************************************************
localhost                  : ok=4    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

Reply:
I have added your suggestion. While it is technically not correct, since we want to adress the result of the first (0) query and nothing else, this simplifies the code and is intuitively understandable by users.
Your suggestion also opens the path to combine the two tasks which collect the users depending on mysql version. We can supply the `mysql_query` module with two separate queries in one task and we would get two arrays, that are flattened in this step. This might simplify things even further. At least we could drop one additional variable and a lot of conditions. How do you feel about that?

Reply:
> While it is technically not correct, since we want to adress the result of the first (0) query and nothing else

Are there any ways more than one query-result could be returned, aside from executing multiple queries?

> How do you feel about that?

I do like it!

Reply:
Okay, so we keep it the way it is right now?

Reply:
I'd say yes. This PR deals with improving the current implementation and I just created an issue with the improvement of the general approach.